### PR TITLE
Remove support for Ubuntu 22.04

### PR DESCRIPTION
**Pull request by derrod**
### Description

Remove Ubuntu 22.04 support and workarounds. Update CI to 24.04.

### Motivation and Context

22.04 is old and having to deal with its old dependencies (especially Qt) is becoming increasingly burdensome.

### How Has This Been Tested?

Hasn't.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)
- Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by Warchamp7** [comment]
Noting that we have planned and communicated to do this 12~ months after 24.04 which would be April

https://github.com/obsproject/obs-studio/discussions/9055

**Review by RytoEX — CHANGES_REQUESTED** [review]
Might be one other place to remove the jansson macro.

**Review by PatTheMav — CHANGES_REQUESTED** [review]
You forgot to remove the lower compatible FFmpeg target in `plugins/obs-ffmpeg/cmake/dependencies.cmake`.

File: UI/update/models/whatsnew.hpp [hunk 1]
```diff
@@ -21,29 +21,6 @@
...
-	}                                                                   \
-	friend void from_json(const nlohmann::json &nlohmann_json_j,        \
-			      Type &nlohmann_json_t)                        \
-	{                                                                   \
-		Type nlohmann_json_default_obj;                             \
-		NLOHMANN_JSON_EXPAND(NLOHMANN_JSON_PASTE(                   \
-			NLOHMANN_JSON_FROM_WITH_DEFAULT, __VA_ARGS__))      \
-	}
-
-#endif
```
[the remark this answers is not in the document]
Comment (derrod) [reply UI/update/models/whatsnew.hpp:45]: Yep, this was copied there too.

**Comment by derrod** [comment]
> You forgot to remove the lower compatible FFmpeg target in `plugins/obs-ffmpeg/cmake/dependencies.cmake`.

Bumped the minimum to `6.1` there and in the libobs `CMakeLists.txt`.

**Review by RytoEX — APPROVED** [review]
Tentatively LGTM

**Review by Gol-D-Ace — APPROVED** [review]
Like already discussed internally I'm fine with ending support for 22.04 now because people should or will very shortly have an official upgrade path from 22.04 to 24.04.

We also offer alternatives (Flatpak) in case you don't want to upgrade your LTS yet but still prefer to have new obs features

File: .github/scripts/utils.zsh/setup_ubuntu [hunk 2]
```diff
@@ -101,7 +101,8 @@ sudo apt-get install -y --no-install-recommends \
   libasound2-dev libfdk-aac-dev libfontconfig-dev libfreetype6-dev libjack-jackd2-dev \
   libpulse-dev libsndio-dev libspeexdsp-dev libudev-dev libv4l-dev libva-dev libvlc-dev \
   libpci-dev libdrm-dev \
-  nlohmann-json3-dev libwebsocketpp-dev libasio-dev libqrcodegencpp-dev
+  nlohmann-json3-dev libwebsocketpp-dev libasio-dev libqrcodegencpp-dev \
+  libffmpeg-nvenc-dev
```
[the remark this answers is not in the document]
Comment (Gol-D-Ace) [reply .github/scripts/utils.zsh/setup_ubuntu:105]: From my memory they get handled by ffmpeg now. At least I don't have them in my PPA config either.

[the remark this answers is not in the document]
Comment (RytoEX) [reply .github/scripts/utils.zsh/setup_ubuntu:105]: This does seem to suggest that libavformat depends on librist and libsrt, unless I'm misreading it.
https://packages.ubuntu.com/noble/libavformat60

[the remark this answers is not in the document]
Comment (RytoEX) [reply .github/scripts/utils.zsh/setup_ubuntu:105]: In any case, there should be no harm in explicitly moving the `UBUNTU_2210_OR_LATER` section into this section.

[the remark this answers is not in the document]
Comment (derrod) [reply .github/scripts/utils.zsh/setup_ubuntu:105]: The setup script already installs those for recent enough Ubuntu versions, though we could move these items to the main apt command now.

https://github.com/obsproject/obs-studio/blob/46b5f7e6d079f06f2a0ca426d89b429c5013e097/.github/scripts/utils.zsh/setup_ubuntu#L108

[the remark this answers is not in the document]
Comment (RytoEX) [reply .github/scripts/utils.zsh/setup_ubuntu:105]: > though we could move these items to the main apt command now.

I think that's the suggestion.

[the remark this answers is not in the document]
Comment (PatTheMav) [reply .github/scripts/utils.zsh/setup_ubuntu:105]: Yep I couldn't find the `UBUNTU_2210_OR_LATER` in the changed files, but that's what I was aiming at - the whole check is unnecessary with a move to 24.04.

[the remark this answers is not in the document]
Comment (derrod) [reply .github/scripts/utils.zsh/setup_ubuntu]: Updated the scripts

**Review by RytoEX — APPROVED** [review]
LGTM

File: .github/scripts/utils.zsh/setup_ubuntu (hunk 2 above)
[the remark this answers is not in the document]
Comment (RytoEX) [reply .github/scripts/utils.zsh/setup_ubuntu]: The update looks good to me.

[the remark this answers is not in the document]
Comment (RytoEX) [reply .github/scripts/utils.zsh/setup_ubuntu]: Going to make the call to resolve this feedback thread.

**Pull request merged**